src/ucp/core/ucp_context.h

```diff
@@ -348,8 +348,8 @@ const char* ucp_tl_bitmap_str(ucp_context_h context, uint64_t tl_bitmap,
 const char* ucp_feature_flags_str(unsigned feature_flags, char *str,
                                   size_t max_str_len);
 
-ucs_memory_type_t
-ucp_memory_type_detect_mds(ucp_context_h context, const void *address, size_t length);
+ucs_mem_info_t
+ucp_mem_info_detect_mds(ucp_context_h context, const void *address, size_t length);
```

Comment:
I think better to return nothing (`void`) and pass the pointer to `ucs_mem_info_t` to avoid copying from the return value to a local variable
wdyt?

Reply:
> 
> 
> > @bureddy @yosefe
> > Compilation failed because `ucs/sys/topo.h` wasn't found. Any thoughts on this? I didn't run into this issue with manual builds at my end.
> > https://dev.azure.com/ucfconsort/ucx/_build/results?buildId=4864&view=logs&jobId=1b6e0548-13f3-5bbf-46b7-02377d74d406&j=1b6e0548-13f3-5bbf-46b7-02377d74d406&t=a3dbc0da-a4bc-5096-251c-c5e42b51def6
> 
> @Akshay-Venkatesh since `sys/memory/memory_type.h` is the installed header file, we have to add `ucs/sys/topo.h` to the list of the headers that are installed -
> 
> https://github.com/openucx/ucx/blob/f3de7f3732a8d5c7edcc038b17746503e69903d4/src/ucs/Makefile.am#L38
> 
> btw, it wasn't added to any list (neither `noninst` nor `inst`)

Just did that. Not sure why build didn't fail even though I ran an autogen.sh